app/controllers/api/vcf_files_controller.rb

```diff
@@ -4,8 +4,28 @@ class Api::VcfFilesController < Api::BaseController
   # POST /vcf_files
   def create
     fname = File.basename( params[:file].original_filename)
+    case_id = fname.split(".")[0]
     warnings = ""
     alerts = ""
+    
+    # check if the corresponding case is already created
+    p = Patient.where(case_id:case_id).first
```

Comment:
case id is unique. You can use find or find_by instead of using where and taking the first one.